src/System.Management.Automation/engine/Utils.cs

```diff
@@ -482,9 +482,14 @@ internal static string GetWindowsPowerShellVersionFromRegistry()
 
         internal static string GetApplicationBase(string shellId)
         {
-            // Use the location of SMA.dll as the application base.
-            Assembly assembly = typeof(PSObject).Assembly;
-            return Path.GetDirectoryName(assembly.Location);
+            var baseDirectory = Path.GetDirectoryName(typeof(PSObject).Assembly.Location);
```

Comment:
```suggestion
            // Use the location of SMA.dll as the application base if it exists,
            // otherwise, use the base directory from `AppContext`.
            var baseDirectory = Path.GetDirectoryName(typeof(PSObject).Assembly.Location);
```